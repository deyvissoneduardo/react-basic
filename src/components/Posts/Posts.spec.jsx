import React from "react";
import { Posts } from ".";
import { render, screen } from "@testing-library/react";

const props = {
  posts: [
    {
      id: 1,
      title: "title 01",
      body: "body01",
      cover: "img/img01.png",
    },
    {
      id: 2,
      title: "title 02",
      body: "body02",
      cover: "img/img02.png",
    },
    {
      id: 3,
      title: "title 03",
      body: "body03",
      cover: "img/img03.png",
    },
  ],
};

describe("<Posts />", () => {
  test("should render posts", () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole("img", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  test("should match snapshot", () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
