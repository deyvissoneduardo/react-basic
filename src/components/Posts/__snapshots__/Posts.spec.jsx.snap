// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 01"
      src="img/img01.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title 01
      </h1>
      <p>
        body01
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 02"
      src="img/img02.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title 02
      </h1>
      <p>
        body02
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 03"
      src="img/img03.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title 03
      </h1>
      <p>
        body03
      </p>
    </div>
  </div>
</div>
`;
